cmd/soroban-cli/src/utils.rs

```diff
@@ -156,10 +159,17 @@ pub fn sign_transaction(
 /// # Errors
 ///
 /// Might return an error
-pub fn id_from_str<const N: usize>(contract_id: &String) -> Result<[u8; N], FromHexError> {
-    padded_hex_from_str(contract_id, N)?
-        .try_into()
-        .map_err(|_| FromHexError::InvalidStringLength)
+pub fn contract_id_from_str(contract_id: &str) -> Result<[u8; 32], stellar_strkey::DecodeError> {
+    stellar_strkey::Contract::from_string(contract_id)
+        .map(|strkey| strkey.0)
+        .or_else(|_| {
+            // strkey failed, try to parse it as a hex string, for backwards compatibility.
```

Comment:
👏🏻 Very nice keeping the backwards compatibility.

Reply:
Should we make a note to drop the deprecated way of doing things prior to v1.0?